src/app/qgsmapsavedialog.cpp

```diff
@@ -92,6 +107,12 @@ QgsMapSaveDialog::QgsMapSaveDialog( QWidget *parent, QgsMapCanvas *mapCanvas, co
 
     this->setWindowTitle( tr( "Save map as PDF" ) );
   }
+
+  connect( buttonBox, &QDialogButtonBox::accepted, this, &QgsMapSaveDialog::accepted );
+  connect( buttonBox, &QDialogButtonBox::rejected, this, [ this ]
+  {
+    delete this;
```

Comment:
@m-kuhn , I was doing that before, but it messed up with the QPrinter* / QImage* / QPainter* objects (as well as killing the connected signals too early, which resulted in the failure to show "successful export" message bar).